src/main/java/org/pentaho/di/trans/dataservice/optimization/OptimizationImpactInfo.java

```diff
@@ -76,7 +76,7 @@ public String getDescription() {
       .append( "Step:  " )
       .append( getStepName() )
       .append( "\n" );
-    if ( getErrorMsg().length() > 0 ) {
+    if ( getErrorMsg() == null || getErrorMsg().length() > 0 ) {
       builder
```

Comment:
@denis8703 This check will result in an [ERROR] to be logged if the error message is null, which may not be correct.  I think it's probably better to assume that an empty or null error message means "no error".